src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -757,6 +757,10 @@ private static Repository linkRepositoryForExistingGit(Path localPath, VcsReposi
             gitRepoConfig.setString(ConfigConstants.CONFIG_BRANCH_SECTION, defaultBranch, ConfigConstants.CONFIG_REMOTE_SECTION, REMOTE_NAME);
             gitRepoConfig.setString(ConfigConstants.CONFIG_BRANCH_SECTION, defaultBranch, ConfigConstants.CONFIG_MERGE_SECTION, "refs/heads/" + defaultBranch);
 
+            RefUpdate refUpdate = repository.getRefDatabase().newUpdate(Constants.HEAD, false);
```

Comment:
we should add a comment that this is (apparently?) important when creating new repositories or when dealing with empty repositories?